source/coap_connection_handler.c

```diff
@@ -196,12 +196,16 @@ static void secure_recv_sckt_msg(void *cb_res);
 
 static internal_socket_t *int_socket_create(uint16_t listen_port, bool use_ephemeral_port, bool is_secure, bool real_socket, bool bypassSec)
 {
+    uint8_t ll_security = 0;
```

Comment:
The option takes an int8_t. How about just:

    int8_t ll_security = !bypassSec;

or

    int8_t ll_security = bypassSec ? 0 : 1;

Although that's still inconsistent with the other socket option which uses a compound literal. To match that, it would be:

    socket_setsockopt(this->listen_socket, SOCKET_IPPROTO_IPV6, SOCKET_LINK_LAYER_SECURITY, 
                                &(const int8_t) {
                                    bypassSec ? 0 : 1
                                }, sizeof(int8_t));

Not sure that's neater though.